drivers/serial/uart_mspm0.c

```diff
@@ -37,64 +38,223 @@ struct uart_mspm0_data {
-	DL_UART_reset(config->regs);
-	DL_UART_enablePower(config->regs);
-	delay_cycles(POWER_STARTUP_DELAY);
-
-	/* Init UART pins */
-	ret = pinctrl_apply_state(config->pinctrl, PINCTRL_STATE_DEFAULT);
-	if (ret < 0) {
-		return ret;
-	}
-
 	/* Set UART configs */
 	DL_UART_setClockConfig(config->regs, (DL_UART_ClockConfig *)&data->UART_ClockConfig);
 	DL_UART_init(config->regs, (DL_UART_Config *)&data->UART_Config);
 
 	/*
 	 * Configure baud rate by setting oversampling and baud rate divisor
-	 * from the device tree data current-speed
+	 * from the selected current-speed
 	 */
 	ret = clock_control_get_rate(clk_dev, (clock_control_subsys_t)config->clock_subsys,
 				     &clock_rate);
 	if (ret < 0) {
 		return ret;
 	}
 
-	DL_UART_configBaudRate(config->regs, clock_rate, config->current_speed);
+	DL_UART_Main_configBaudRate(config->regs, clock_rate, data->current_speed);
 
-#ifdef CONFIG_UART_INTERRUPT_DRIVEN
-	config->irq_config_func(dev);
-#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+	return 0;
+}
 
-	/* Enable UART */
-	DL_UART_enable(config->regs);
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+
+uint32_t uart_parity_to_mspm0[5] = {
+	DL_UART_MAIN_PARITY_NONE,
+	DL_UART_MAIN_PARITY_ODD,
+	DL_UART_MAIN_PARITY_EVEN,
+	DL_UART_MAIN_PARITY_STICK_ONE,
+	DL_UART_MAIN_PARITY_STICK_ZERO,
+};
+
+uint32_t uart_stop_bits_to_mspm0[4] = {
+	UINT32_MAX,
+	DL_UART_MAIN_STOP_BITS_ONE,
+	UINT32_MAX,
+	DL_UART_MAIN_STOP_BITS_TWO,
+};
+
+uint32_t uart_data_bits_to_mspm0[4] = {
+	DL_UART_MAIN_WORD_LENGTH_5_BITS,
+	DL_UART_MAIN_WORD_LENGTH_6_BITS,
+	DL_UART_MAIN_WORD_LENGTH_7_BITS,
+	DL_UART_MAIN_WORD_LENGTH_8_BITS,
+};
+
+uint32_t uart_flow_control_to_mspm0[2] = {
+	DL_UART_MAIN_FLOW_CONTROL_NONE,
+	DL_UART_MAIN_FLOW_CONTROL_RTS_CTS,
+};
+
+static int uart_mspm0_translate_in(uint32_t value_array[],
+				   int value_array_length,
+				   uint8_t uart_cfg_value,
+				   uint32_t *mspm0_cfg_value)
+{
+	if (uart_cfg_value >= value_array_length) {
+		return -EINVAL;
+	}
+
+	if (value_array[uart_cfg_value] == UINT32_MAX) {
+		return -ENOSYS;
+	}
+
+	*mspm0_cfg_value = value_array[uart_cfg_value];
+
+ 	return 0;
+}
+
+static int uart_mspm0_transtlate_out(uint32_t value_array[],
+				     int value_array_length,
+				     uint32_t mspm0_cfg_value,
+				     uint8_t *uart_cfg_value)
+{
+	int idx;
+
+	for (idx = 0; idx < value_array_length; idx++) {
+		if (value_array[idx] == mspm0_cfg_value) {
+			break;
+		}
+	}
+
+	if (idx == value_array_length ||
+	    value_array[idx] == UINT32_MAX) {
+		return -EINVAL;
+	}
+
+	*uart_cfg_value = (uint8_t)idx;
 
 	return 0;
 }
 
-static int uart_mspm0_poll_in(const struct device *dev, unsigned char *c)
+static int uart_mspm0_configure(const struct device *dev,
+				const struct uart_config *cfg)
 {
 	const struct uart_mspm0_config *config = dev->config;
+	struct uart_mspm0_data *data = dev->data;
+	uint32_t value;
+	int ret;
 
-	return (DL_UART_receiveDataCheck(config->regs, c)) ? 0 : -1;
+	DL_Main_disable(config->regs);
```

Comment:
This does not seem to exist. Should probably be `DL_UART_disable` (same for enable)

Reply:
> This does not seem to exist. Should probably be `DL_UART_disable` (same for enable)

I tested the driver on 4.2 (as it's a 1:1 with upstream),
our driver in 4.1.0 is still mostly using this _Main versions of functions... blindly did a rename

Reply:
I am saying shit (in addition of sending a non-compiled version of a PR, what can go wrong switching back from 4.2 to 4.1...):  we should use _Main versions. 
and yes it's' DL_UART_Main_disable